docs/config/templates/ngdoc/api/directive.template.html

```diff
@@ -29,11 +29,22 @@ <h2 id="usage">Usage</h2>
       </li>
     {% endif -%}
 
+
+    {% set hasNameAsParam = false %}
+
+    {# when a directive's name is not a parameter (i.e. doesn't take a value),
+      add the directive name to the list of attributes and/or css classes #}
+
+    {%- for param in doc.params %}
+    {% set hasNameAsParam = true if param.name === doc.name else hasNameAsParam %}
+    {%- endfor %}
+
     {%- if doc.restrict.attribute -%}
       <li>as attribute:
         {% code %}
-        <{$ doc.element $}
-          {%- for param in doc.params %}
+        <{$ doc.element $} {%- if not hasNameAsParam %}
```

Comment:
given that you have used `{% -` to ignore the whitespace, you may as well start this `if` block on the next line.

Reply:
Something like:

```
        <{$ doc.element $}
          {%- if not hasNameAsParam %}
          {$ lib.directiveParam(param.name, param.type, '="', '"') $}
          {% endif %}
          {%- for param in doc.params %}
          {$ lib.directiveParam(param.name, param.type, '="', '"') $}
          {%- endfor %}>

```

Reply:
Thanks - I still need to include `{%- endif -%}` because otherwise there's a blank line after the directive name when hasNameAsParam is true.